### PR TITLE
fix `parentfile`

### PR DESCRIPTION
**Pull request by t-bltg**
`parentfile` is inconsistent with `filedata`: on a local build of julia, where `juliadir` points to `basebuilddir`, `parentfile` returns a path starting with `Sys.BINDIR/Base.DATAROOTDIR` because of `Base.find_source_file`.

This is not consistent with `filedata` (`Base._included_files`) which point to files in `basebuilddir`.

- fix https://github.com/timholy/Revise.jl/issues/697;
- add a specific ci test using a forced `juliadir` path (`test/juliadir.jl`).
- update `ci.yaml`, use `@latest` if possible, rewrite some parts for clarity;
- use `julia-actions/cache@v1`;

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -54,21 +45,41 @@ jobs:
...
-        julia --project --code-coverage=user -e 'ENV["JULIA_REVISE_POLL"]="1"; using Pkg, Revise; include(joinpath(dirname(pathof(Revise)), "..", "test", "polling.jl"))'
+        julia --project --code-coverage=user -e '
+          ENV["JULIA_REVISE_POLL"]="1"
+          using Pkg, Revise
+          include(joinpath(dirname(pathof(Revise)), "..", "test", "polling.jl"))
+        '
         # The REPL wasn't initialized, so the "Methods at REPL" tests didn't run. Pick those up now.
-        TERM="xterm" julia --project --code-coverage=user -e 'using InteractiveUtils, REPL, Revise; @async(Base.run_main_repl(true, true, false, true, false)); while !isdefined(Base, :active_repl_backend) sleep(0.1) end; pushfirst!(Base.active_repl_backend.ast_transforms, Revise.revise_first); include(joinpath("test", "runtests.jl")); REPL.eval_user_input(:(exit()), Base.active_repl_backend)' "Methods at REPL"
+        TERM="xterm" julia --project --code-coverage=user -e '
+          using InteractiveUtils, REPL, Revise
```
Comment (t-bltg) [inline .github/workflows/ci.yml:55]: In bash, single (strong) quotes can span multiple lines, this is clearer than a single line statement with semicolons.

File: test/juliadir.jl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using Revise, InteractiveUtils, Test
```
Comment (timholy) [inline test/juliadir.jl:1]: Should this be executed from `runtests.jl`?

**Comment by timholy** [comment]
Thanks!

File: test/juliadir.jl (hunk 2 above)
Comment (t-bltg) [reply test/juliadir.jl:1]: Yes I thought it was relevant to add it to 'extra test' section in ci, because we download the julia source files for that version, and force `juliadir` to mimic `basebuilddir` for testing. Since it's a bit of a corner case, and hackery machinery is involved, I think it's relevant to split it from `runtests.jl`.

**Comment by t-bltg** [comment]
Thanks for merging ;)
